### ISSUE TITLE
add an "--add-assert-to-assume" option

### ISSUE THREAD

**Issue by Laplace-Demon** [OCamlPro/owi#359]
When an assertion is done, add it to assume to boost symbolic execution

**Comment by Laplace-Demon** [comment]
added it in webassembly rac mode
123935903ed81999430c55154f2f903d4b6ecd89

**Comment by zapashcanon** [comment]
Actually, it would be something more like "adding assert to PC"

**Comment by zapashcanon** [comment]
I am also tempted to think this should be the default

**Comment by Laplace-Demon** [comment]
I agree, I think it's mostly better to have previous assertions available in path conditions.

If I recall right, this is an option of Weasel code generator, and Owi primitives are not exposed to the Wasm file. In that case, having `assert_and_hint` seems to only serve internal use, and does not make much difference. But I think it may be interesting in the setting where the user call `owi_assert` directly? (maybe in the C examples, I'm not sure)

**Comment by zapashcanon** [comment]
> But I think it may be interesting in the setting where the user call owi_assert directly? 

Yes, that's mostly what I'm thinking of. It would be useful to add to the PC to help solvers, but sometimes it may also hurt performances so we should let the user decide and try to find what's the best default.
